pkg/destroy/aws/aws.go

```diff
@@ -361,7 +366,7 @@ func getSharedHostedZone(client *route53.Route53, privateID string, logger logru
 			if *zone.Name != privateName {
```

Comment:
Now that we are allowing input zones of unknown privacy through here as well (vs. requiring them to be private), we probably want a guard against ID matching.

```go
if *zone.Id == privateID {
  continue
}

if *zone.Name != privateName {
```

Reply:
> Do you have an opinion about whether we should process input zones of unknown privacy or strictly zones for which we can confirm the privacy?

If the input zone has indeterminate privacy and we find a known-public zone with the same name, I think that's sufficient grounds for assuming the caller gave us a private zone and going ahead to delete entries from the discovered public zone.